src/bus/types.rs

```diff
@@ -30,16 +30,31 @@ pub enum AddressSecretKey {
     #[display("addr_key({address}, ..)")]
     Bitcoin {
         address: bitcoin::Address,
-        secret_key: bitcoin::secp256k1::SecretKey,
+        secret_key_info: BitcoinSecretKeyInfo,
     },
     #[display("addr_key({address}, ..)")]
     Monero {
         address: monero::Address,
-        view: monero::PrivateKey,
-        spend: monero::PrivateKey,
+        secret_key_info: MoneroSecretKeyInfo,
     },
 }
 
+#[derive(Clone, Debug, Display, Eq, PartialEq, NetworkDecode, NetworkEncode)]
+#[display("bitcoin secret key info")]
+pub struct BitcoinSecretKeyInfo {
+    pub swap_id: Option<SwapId>,
```

Comment:
I don't understand why this is an option, and not just the naked type? We only instantiate secret keys for both sides in the context of a swap. The only case where this ain't `Some(..)` is inside the [lmdb test](https://github.com/TheCharlatan/farcaster-node/blob/cad412a782ee6755481c538c003a3f901b8c0db9/src/databased/runtime.rs#L692).

Reply:
I made this an Option for flexibility purposes, we might have a secret key in the future that is not attached to a swap id. It's a _very_ speculative reason, so if that's not good enough for you, let me know.

Reply:
Then it's equivalent to some _NULLABLE_ column in other type of db, which is fine by me if it helps us avoid breaking the storage format later. I'd just add a comment on the struct to avoid confusion in the future (not related to protocol, just storage layout)

Reply:
thanks for the rationale - I'm fine to keep the speculative flexibility to drop swap_id coupling: I have a bunch of checkpoints that are useless at this stage (unless I go through the pain of rebuilding the old fc versions, or even just figuring out which ones those were), so let's not risk more of that for users ;)